invokeai/frontend/src/features/parameters/components/ProcessButtons/CancelButton.tsx

```diff
@@ -49,15 +66,50 @@ export default function CancelButton(
     [isConnected, isProcessing, isCancelable]
   );
 
+  useEffect(() => {
+    if (cancelAfterIteration < currentIteration) {
+      handleClickCancel();
+    }
+  }, [cancelAfterIteration, currentIteration, handleClickCancel]);
+
   return (
-    <IAIIconButton
-      icon={<MdCancel />}
-      tooltip={t('parameters:cancel')}
-      aria-label={t('parameters:cancel')}
-      isDisabled={!isConnected || !isProcessing || !isCancelable}
-      onClick={handleClickCancel}
-      styleClass="cancel-btn"
-      {...rest}
-    />
+    <>
+      <IAIIconButton
+        icon={<MdCancel />}
+        tooltip={t('parameters:cancel')}
+        aria-label={t('parameters:cancel')}
+        isDisabled={!isConnected || !isProcessing || !isCancelable}
+        onClick={handleClickCancel}
+        styleClass="cancel-btn"
+        {...rest}
+      />
+      <IAIIconButton
+        icon={isCancelScheduled ? <Spinner /> : <MdCancelScheduleSend />}
```

Comment:
Let's use Chakra's `ButtonSpinner` component here instead of `Spinner` (it is the spinner that is displayed when `isLoading` is used).